### PR TITLE
Upgrading notes for user profile

### LINKED ISSUE #24491: Upgrading notes for user profile
### Description

We might need to add some upgrading notes for user profile in this release.

### PR DESCRIPTION
**Pull request by mposolda**
closes #24491

### CODE REVIEW

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 1]
```diff
@@ -57,5 +57,6 @@ paginated lookup of subgroups. Thanks to https://github.com/alice-wondered[Alice
 
 Declarative user profile is still a preview feature in this release, but we are working hard on promoting it to a supported feature. Feedback is welcome.
 If you find any issues or have any improvements in mind,  you are welcome to create https://github.com/keycloak/keycloak/issues/new/choose[Github issue],
-ideally with label `area/user-profile`.
+ideally with label `area/user-profile`. It is also recommended to check link:{upgradingguide_link}[{upgradingguide_name}]  with the migration changes for this
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:60]:
```suggestion
ideally with the label `area/user-profile`. It is also recommended to check the link:{upgradingguide_link}[{upgradingguide_name}]  with the migration changes for this
```

File: docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc [hunk 2]
```diff
@@ -141,3 +141,12 @@ Relying on RESTEasy Classic is not longer an option because it is not available
...
+
+= User Profile updates
+
+There are lots of fixes and updates related to user profile as we are working on promoting this feature from preview to officially supported.
+There are minor changes to the SPI like newly added method `boolean isEnabled(RealmModel realm)` on `UserProfileProvider` interface. Also
+some user profile classes as well as some validator related classes (but not builtin validator implementations) were moved from `keycloak-server-spi-private` to
+`keycloak-server-spi` module. However the packages for java classes remain the same. You might be affected just in some corner cases, like for instance when you
+are overriding the built-in implementation with your own `UserProfileProvider` implementation (however note that `UserProfileProvider` is unsupported SPI).
```
Comment (andymunro) [inline docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc:151]:
```suggestion
This release includes many fixes and updates that are related to user profile as we are working on promoting this feature from preview to officially supported.
Minor changes exist for the SPI such as the newly added method `boolean isEnabled(RealmModel realm)` on `UserProfileProvider` interface. Also
some user profile classes and some validator related classes (but not builtin validator implementations) were moved from `keycloak-server-spi-private` to
`keycloak-server-spi` module. However, the packages for java classes remain the same. You might be affected in some corner cases, such as when you
are overriding the built-in implementation with your own `UserProfileProvider` implementation However, note that `UserProfileProvider` is an unsupported SPI.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
A few suggestions, @mposolda

**Comment by mposolda** [comment]
@andymunro Thanks!

**Review by pedroigor — APPROVED** [review]

**Pull request merged**